core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -2398,4 +2398,11 @@ package object config {
       .version("3.3.0")
       .intConf
       .createWithDefault(5)
+
+  private[spark] val HEARTBEAT_RECEIVER_CHECK_WORKER_LAST_HEARTBEAT =
+    ConfigBuilder("spark.driver.heartbeat.checkWorkerLastHeartbeat")
+      .internal()
+      .version("3.4.0")
+      .booleanConf
+      .createWithDefault(true)
```

Comment:
Default this to `false`

Reply:
I will update the default value to false when this PR is ready and passes CI/CD tests.

Reply:
Please update the relevant tests, so that they test with this being `true`/`false` as relevant - instead of a one-time CI/CD pass : so that future evolution of the code will continue to test this feature (with and without the flag)

Reply:
Updated. https://github.com/apache/spark/pull/37411/commits/bb5c2193962405f50242d7728a5cc6529d7eb229